ext/standard/file.c

```diff
@@ -1487,26 +1489,16 @@ PHP_FUNCTION(ftruncate)
 	RETURN_BOOL(0 == php_stream_truncate_set_size(stream, size));
 }
 /* }}} */
-
-/* {{{ Stat() on a filehandle */
-PHP_FUNCTION(fstat)
+PHPAPI void php_fstat(php_stream *stream, zval *return_value)
```

Comment:
Should I make this return a zval* instead of the out param?

Reply:
How is failure signaled?

Reply:
It would be signalled by a IS_FALSE zval* as it's currently the value returned when fstats() fails. Not sure if that's a good idea but signalling it via ``SUCCESS/FAILURE`` or ``true/false`` seems rather pointless because I would expect anyone using this API to want the result of the fstats call if it succeeded.

Reply:
Ah, thanks! If I understand correctly, `return_value` is an in/out parameter, and that would be fine.